server/src/weaverbird/backends/mongo_translator/steps/convert.py

```diff
@@ -0,0 +1,87 @@
+
+from weaverbird.backends.mongo_translator.steps.types import MongoStep
+from weaverbird.pipeline.steps import ConvertStep
+
+TYPE_MAP = {
+    'boolean': 'bool',
+    'date': 'date',
+    'float': 'double',
+    'integer': 'long',  # better to cast into 64-bit integers. 32-bit integers ('int') cannot be casted to dates
+    'text': 'string',
+}
+
+
+def translate_convert(step: ConvertStep) -> List[MongoStep]:
+    mongo_add_fields = {}
+    mongo_type = TYPE_MAP[step.data_type] if step.data_type in TYPE_MAP else ''
+    for column in step.columns:
+        _input: Union[dict, str]
+        _on_null: Optional[str] = None
+
+        if mongo_type == 'date':
+            _input = {
+                '$switch': {
+                    'branches': [
+                        # Mongo cannot manage string representing only a year (e.g: "2020")
+                        # so we need to concatenate the month and day ("-01-01"):
+                        {
+                            'case': {
+                                '$and': [
+                                    {'$eq': [{'$type': f'${column}'}, 'string']},
+                                    {'$eq': [{'$strLenCP': f'${column}'}, 4]},
+                                ]
+                            },
+                            'then': {'$concat': [f'${column}', '-01-01']},
+                        },
+                        # Mongo cannot cast integers into date but only long into date,
+                        # so we manage the cast from int to long when needed:
+                        {
+                            'case': {'$eq': [{'$type': f'${column}'}, 'int']},
+                            'then': {'$toLong': f'${column}'},
+                        },
+                    ],
+                    'default': f'${column}',
+                }
+            }
+        elif mongo_type == 'long':
+            # First, convert column to double, then to long.
+            # Otherwise, '12.123' cannot be converted to an int, and gives an error.
+            _input = {
+                '$convert': {
+                    'input': f'${column}',
+                    'to': 'double',
+                    'onError': None,
+                }
+            }
+        elif mongo_type == 'bool':
+            # Force conversion of empty strings to 'False' (otherwise it gives True)
+            _input = {
+                '$switch': {
+                    'branches': [{'case': {'$eq': [f'${column}', '']}, 'then': False}],
+                    'default': f'${column}',
+                }
+            }
+        elif mongo_type == 'string':
+            # First, convert NaN to null:
+            _input = {
+                '$switch': {
+                    'branches': [{'case': {'$eq': [f'${column}', float('NaN')]}, 'then': None}],
+                    'default': f'${column}',
+                }
+            }
+            # Then, null will be converted to the literal string 'nan':
+            _on_null = 'nan'
```

Comment:
This is a specific pandas' behavior to use the string 'nan'.
Maybe we would think than `null` / `None` is more appropriate? (because we can filter on it quickly after, and it denotes some missing data)
Maybe that's something we'd like to do differently than pandas?